prov/gni/src/gnix_mr_cache.c

```diff
@@ -445,6 +445,10 @@ __clear_notifier_events(gnix_mr_cache_t *cache)
 		return;
 	}
 
+	if (!cache->attr.notifier) {
```

Comment:
I'd check this before lazy dereg for now.  I say *for now*, because it's possible that the lazy dereg check will go away or be refactored.